src/orion/core/io/config.py

```diff
@@ -451,3 +451,35 @@ def to_dict(self):
                 config[key] = self[key].to_dict()
 
         return config
+
+    def from_dict(self, config):
+        """Set the configuration from a dictionary"""
+
+        logger.debug("Setting config to %s", config)
+        logger.debug("Config was %s", repr(self))
+
+        with _disable_logger():
+            for key in self._config:  # pylint: disable=consider-using-dict-items
+                value = config.get(key)
+
+                if value:
```

Comment:
Seems like this would miss 0 and '' items.

Reply:
fixed